src/ct_eval.cpp

```diff
@@ -250,14 +250,15 @@ TypedAST::DeclarationList* ct_eval(
 
 	for (auto& decl : ast->m_declarations) {
 		int meta_type = tc.m_core.m_meta_core.find(decl.m_meta_type);
-		// TODO: ban type hints from type function and mono type declarations?
 		if (meta_type == tc.meta_typefunc()) {
+			assert(!decl.m_type_hint && "type hint not allowed in type function declaration");
```

Comment:
I feel like this check should be in metacheck, and not here. Thoughts?

Reply:
I thought so as well, but it seemed like much less work to put them there hahaha. Perhaps we will fix that when we clean up.